includes/Core/Util/User_Input_Settings.php

```diff
@@ -0,0 +1,206 @@
+		User_Transients $user_transients = null
+	) {
+		$this->transients      = $transients ?: new Transients( $context );
+		$this->user_transients = $user_transients ?: new User_Transients( $context );
+		$this->authentication  = $authentication ?: new Authentication( $context, null, null, $this->transients );
+	}
+
+	/**
+	 * Determines whether the site is connected to proxy or not.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return boolean TRUE if connected, otherwise FALSE.
+	 */
+	private function is_connected_to_proxy() {
+		if ( ! $this->authentication->is_authenticated() ) {
+			return false;
+		}
+
+		$credentials = $this->authentication->credentials();
+		if ( $credentials ) {
+			if ( ! $credentials->using_proxy() || ! $credentials->has() ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Sends POST request to the proxy's settings endpoint to sync user input settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $settings User settings.
+	 * @return array User input settings.
+	 */
+	private function sync_with_proxy( $settings ) {
+		$user_input_settings_url  = $this->authentication->get_google_proxy()->url( Google_Proxy::USER_INPUT_SETTINGS_URI );
+		$user_input_settings_args = array(
+			'headers' => array(
+				'Authorization' => 'Bearer ' . $this->authentication->get_oauth_client()->get_access_token(),
+			),
+		);
+
+		if ( ! empty( $settings ) ) {
+			$user_input_settings_args['headers']['Content-Type'] = 'application/json';
+			$user_input_settings_args['body']                    = wp_json_encode( $settings );
+		}
+
+		$response = wp_remote_post( $user_input_settings_url, $user_input_settings_args );
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$code = wp_remote_retrieve_response_code( $response );
+		if ( $code < 200 || 299 < $code ) {
+			return new \WP_Error(
+				'user_input_settings_request',
+				__( 'User input settings request failed.', 'google-site-kit' ),
+				array( 'status' => $code )
+			);
+		}
+
+		$body     = wp_remote_retrieve_body( $response );
+		$settings = json_decode( $body, true );
+		if ( is_array( $settings ) ) {
+			$site_settings = array();
+			$user_settings = array();
+
+			foreach ( $settings as $setting_key => $setting_data ) {
+				if ( isset( $setting_data['scope'] ) ) {
+					if ( 'site' === $setting_data['scope'] ) {
+						$site_settings[ $setting_key ] = ! empty( $setting_data['value'] )
+							? $setting_data['value']
+							: array();
+					} elseif ( 'user' === $setting_data['scope'] ) {
+						$user_settings[ $setting_key ] = ! empty( $setting_data['value'] )
+							? $setting_data['value']
+							: array();
+					}
+				}
+			}
+
+			$this->transients->set( 'user_input_settings', $site_settings, WEEK_IN_SECONDS );
+			$this->user_transients->set( 'user_input_settings', $user_settings, WEEK_IN_SECONDS );
+		}
+
+		return $settings;
+	}
+
+	/**
+	 * Gets user input settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array User input settings.
```

Comment:
The return value can also be a `WP_Error`.

Reply:
Updated.